source/funkin/util/FileUtil.hx

```diff
@@ -538,6 +622,145 @@ class FileUtil
     #end
   }
 
+  /**
+   * List all entries in a directory.
+   * Only works on desktop.
+   *
+   * @param path The path to the directory.
+   * @return An array of file and directory names in the directory.
+   */
+  public static function readDir(path:String):Array<String>
+  {
+    #if sys
+    return sys.FileSystem.readDirectory(path);
+    #else
+    return [];
+    #end
+  }
+
+  /**
+   * Move a directory from one location to another, optionally ignoring some paths.
+   * Only works on desktop.
+   *
+   * @param path The path to the directory.
+   * @param destination The path to move the directory to.
+   * @param ignore A list of paths to ignore.
+   */
+  public static function moveDir(path:String, destination:String, ?ignore:Array<String>):Void
+  {
+    #if sys
+    if (!isDirectory(path))
+    {
+      trace('WARNING: Path is not a directory: $path, moving as a file instead');
+      moveFile(path, destination);
+      return;
+    }
+
+    createDirIfNotExists(destination);
+
+    ignore = ignore ?? [];
+    for (entry in readDir(path))
+    {
+      if (ignore.indexOf(Path.join([path, entry])) != -1) continue;
+      if (isDirectory(Path.join([path, entry])))
+      {
+        moveDir(Path.join([path, entry]), Path.join([destination, entry]), ignore);
+      }
+      else
+      {
+        moveFile(Path.join([path, entry]), Path.join([destination, entry]));
+      }
+    }
+
+    if (readDir(path).length == 0)
+    {
+      deleteDir(path);
+    }
+    #else
+    throw 'Direct directory moving by path not supported on this platform.';
+    #end
+  }
+
+  /**
+   * Delete a directory, optionally including its contents, and optionally ignoring some paths.
+   * Only works on desktop.
+   *
+   * @param path The path to the directory.
+   * @param recursive Whether to delete all contents of the directory.
+   * @param ignore A list of paths to ignore.
+   */
+  public static function deleteDir(path:String, recursive:Bool = false, ?ignore:Array<String>):Void
```

Comment:
me when I `deleteDir('../')`

Reply:
that's what we call... **a little trolling** 🤑🤑

Reply:
> me when I `deleteDir('../')`

yeahhh tbh its way too easy... might hold off on this pr until i also figure out how to sandbox file utils

Reply:
![image](https://github.com/user-attachments/assets/9e5b80d3-4da5-441b-b471-f04cb6f8868a)

win